python/google/protobuf/text_format.py

```diff
@@ -469,7 +469,7 @@ def Merge(text,
     ParseError: On text parsing problems.
   """
   return MergeLines(
-      text.split('\n'),
+      re.findall('(?:[^\"\n]|(?:\"(?:[^\"]|[\n])*\"))*\n?', text),
```

Comment:
Yeah. Making people read Regex is a pretty harsh punishment. I added it. Hope it makes sense.